src/components/SearchBox.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { useState } from "react";
+import { makeStyles } from "@material-ui/core";
+import { useHistory } from "react-router-dom";
+import { searchRoute } from "../AppRoutes";
+import classNames from "classnames";
+import TextField from "./TextField";
+
+const useStyles = makeStyles({
+  root: {
+    marginLeft: "auto",
+  },
+});
+
+export default function SearchBox() {
+  const classes = useStyles();
+
+  const [query, setQuery] = useState("");
+
+  const history = useHistory();
+
+  const onSubmit = () => {
+    history.push(`${searchRoute}/${query}`);
```

Comment:
Sorry I saw over it the first time. You have to encode your query.
```suggestion
    history.push(`${searchRoute}/${encodeURIComponent(query)}`);
```